test/passing/js_source.ml.ref

```diff
@@ -9792,4 +9792,4 @@ module Indexop = struct
   let x, y, z = Def.(h.%["one"], h.%("two"), h.%{"three"})
 end
 
-type t = |
+(* type t = | *)
```

Comment:
This change may not be required, we can just assume that we are using 4.07 when running tests ?

Reply:
So you were using 4.06.1, didn't you already get errors on this construct?

Reply:
No.

Reply:
Do you have an idea of what's happening? Why do we have to comment this now? It causes an error on 4.06.1 I don't understand why you didn't have this error previously.

Reply:
I would not expect tests to pass with older version of the compiler.

Reply:
Yes, there are too many moving parts to expect tests to pass with old compilers. My position is just one, heavily influenced by a development context that is perhaps uncommon, and perhaps not widely shared, but I don't think it's worth the time to worry about 4.06 when 4.08 is about to be released. I am much less worried about requiring a current compiler to build ocamlformat than I am about generating code that only the current compiler can parse, but that is a different question. For building ocamlformat itself, I'd say just only develop using the current compiler (to save your own sanity by avoiding chasing down the consequences of the differences between the parsers in different versions) and bump the minimum required version.

Reply:
I agree for the tests, I removed the commit.